tests/datasource/fluent/test_sql_datasources.py

```diff
@@ -28,5 +31,143 @@ def test_kwargs_are_passed_to_create_engine(mocker: MockerFixture):
+        new_callable=mock.PropertyMock,
+        return_value=SQLDatasource(
+            name="my_snowflake_datasource",
+            connection_string="snowflake://<user_login_name>:<password>@<account_identifier>/<database_name>/<schema_name>?warehouse=<warehouse_name>&role=<role_name>",
+        ),
+    ):
+        table_asset = TableAsset(
+            name="my_table_asset",
+            table_name="nonexistent_table_name",
+            schema_name="my_schema",
+        )
+        assert table_asset.table_name not in table_names_in_dbms_schema
+
+
+@pytest.mark.unit
+def test_table_quoted_name_type_all_upper_case_normalizion_is_noop():
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that all upper case entity usage does not undergo any conversion.
+    """
+    table_names_in_dbms_schema: list[str] = [
+        "ACTORS",
+        "ARTISTS",
+        "ATHLETES",
+        "BUSINESS_PEOPLE",
+        "HEALTHCARE_WORKERS",
+        "ENGINEERS",
+        "LAWYERS",
+        "MUSICIANS",
+        "SCIENTISTS",
+        "LITERARY_PROFESSIONALS",
+    ]
+
+    asset_name: str
+    table_name: str
+
+    with mock.patch(
+        "great_expectations.datasource.fluent.sql_datasource.TableAsset.datasource",
+        new_callable=mock.PropertyMock,
+        return_value=SQLDatasource(
+            name="my_snowflake_datasource",
+            connection_string="snowflake://<user_login_name>:<password>@<account_identifier>/<database_name>/<schema_name>?warehouse=<warehouse_name>&role=<role_name>",
+        ),
+    ):
+        for table_name in table_names_in_dbms_schema:
+            asset_name = f"{table_name}_asset"
+            table_asset = TableAsset(
+                name=asset_name,
+                table_name=table_name,
+                schema_name="my_schema",
+            )
+            assert str(table_asset.table_name) == table_name
+            assert str(table_asset.table_name.casefold()) != table_name
+            assert isinstance(table_asset.table_name, sqlalchemy.quoted_name)
+            assert table_asset.table_name.quote is True
+            assert table_asset.table_name in table_names_in_dbms_schema
+
+
+@pytest.mark.unit
+def test_table_quoted_name_type_all_lower_case_normalizion_full():
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that all lower case entity usage undergo conversion to quoted literals.
+    """
+    table_names_in_dbms_schema: list[str] = [
```

Comment:
Same comment about size of list.

Reply:
@anthonyburdi Should I shorten it?

Reply:
@anthonyburdi This has been committed -- thanks.